contracts/RewardManagerFactory.sol

```diff
@@ -0,0 +1,193 @@
+     * @param _cnt cnt token address
+     * @param _startDistribution start timestamp
+     * @param _endDistribution end timestamp
+     * @param _upfrontUnlock Upfront unlock percentage
+     * @param _preMaturePenalty Penalty percentage for pre mature withdrawal
+     * @param _bonusPercentage Bonus rewards percentage for user who hasn't drawn any rewards untill endDistribution
+     * @param _burner Burner for collecting preMaturePenalty
+     * @dev deployer of contract on constructor is set as owner
+     */
+    function launchRewardManager(
+        IERC20 _cnt,
+        uint256 _startDistribution,
+        uint256 _endDistribution,
+        uint256 _upfrontUnlock,
+        uint256 _preMaturePenalty,
+        uint256 _bonusPercentage,
+        address _burner) 
+        public
+        onlyOwner
+    {
+
+        require(address(_cnt) != address(0), "Cant be Zero address");
+        require(address(_burner) != address(0), "Burner Cant be Zero address");
+        
+        require(
+            _startDistribution >= block.timestamp,
+            "Start time should be greater than current"
+        ); // ideally at least 24 hours more to give investors time
+        
+        require(
+            _endDistribution > _startDistribution,
+            "Distribution End Time should be greater than crowdsale StartTime"
+        );
+
+        RewardManager newManager = new RewardManager
+        (
+            _cnt,
+            _startDistribution,
+            _endDistribution,
+            _upfrontUnlock,
+            _preMaturePenalty,
+            _bonusPercentage,
+            _burner
+        );
+
+        managers.push(
+            RewardManagerInfo({ 
+                managerAddress: address(newManager),
+                startDistribution: _startDistribution,
+                endDistribution: _endDistribution
+            })
+        ); //stacking up every crowdsale info ever made to crowdsales variable
+
+        emit RewardManagerLaunched(
+            address(newManager),
+            _startDistribution,
+            _endDistribution
+        );
+        totalRewardManagers++;
+    }
+    
+    function removeRewardManager(uint256 _index) public onlyOwner {
+        require(_index <= managerIndex, "Invalid Index");
+        delete managers[_index];
+    }
+    
+    function userTotalVestingInfo(address _user)
+    public view
+    returns (uint256 totalVested, uint256 totalDrawnAmount, uint256 amountBurnt, uint256 claimable, uint256 bonusRewards, uint256 stillDue){
+        tempInfo memory temp;
+        for(uint256 i = 0; i < totalRewardManagers; i++){
+            address rewardManagerAddress = managers[i].managerAddress;
+            if(rewardManagerAddress != address(0)){
+                RewardManager manager = RewardManager(rewardManagerAddress);
+                (temp._totalVested, temp._totalDrawnAmount, temp._amountBurnt, temp._claimable, temp._bonusRewards, temp._stillDue) = manager.vestingInfo(_user);
```

Comment:
Lets check here if there is non zero amount pending to be claimed. Saving few addition operations for alrady claimed reward managers.